docs/user_authentication/third_party_token_management.md

```diff
@@ -0,0 +1,62 @@
+
+```sequence {theme: 'simple'}
+title Third-party Token Management
+
+participant Pbench-Server
+participant Pbench-Dashboard
+participant Identity-Broker
+participant Identity-Provider
+
+autonumber 1
+Pbench-Dashboard->Pbench-Server: GET Pbench client ID
+
+Pbench-Server->Pbench-Dashboard: 200 Response 
+
+note over Pbench-Server,Pbench-Dashboard:{Identity_Broker_auth_URI: <auth_URI>\nclient_id: <pbench_client_id>\nclient_secret: <pbench_client_secret> # optional\n}
+
+note over Pbench-Dashboard:User clicks login
+
+Pbench-Dashboard->Identity-Broker:LOAD identity broker auth URI\n(Authentication Request)
+
+note right of Pbench-Dashboard:GET request:\n<identity_broker_auth_URI>\n?client_id=<pbench_client_id>\n&response_type=code\n&redirect_uri=<dashboard_URI>\n&scope=openid
+
+note over Identity-Broker:User selects an identity provider from the list
+
+Identity-Broker->Identity-Provider:LOAD identity provider auth page
```

Comment:
Why does this say "LOAD identity provider auth page?"  Is the broker loading the page or is this about a page being loaded into the browser?  If the former, it is not clear to me why we would need to add this to the diagram.  If the latter, should this be the pbench-dashboard?

Perhaps you should have the first participant be the Browser, and call out when the Browser is running dashboard code vs not?

Reply:
I think the browser loads the page after the identity broker makes an appropriate GET request to the identity provider's auth URI. And you are right Pbench is not involved in deciding how that should happen, I am just showing it here at a very high level in case the reader is interested in what happens in the background when the Pbench-dashboard hands the authentication request to the broker.

Reply:
The Pbench Dashboard receives from the Pbench Server the URL for the IDP's authentication page.  When the user clicks the login button, the Dashboard causes the browser to load the IDP login page (providing the IDP with the return URL).  After that point, the Dashboard is not involved (is not running/resident in the browser, as far as I can tell) until the IDP redirects back to the Dashboard page.

So, @portante, if you've got a better way of expressing that, I'd love to hear it....

Reply:
@webbnh I would like to correct a little bit of nomenclature here in your comment, what Pbench-dashboard receives from the Pbench server is the URL for the _Identity_ _broker's_ authentication page, the identity broker may be configured with multiple external IDPs. When the user clicks the login button, the Dashboard causes the browser to load the _Identity_ _broker's_ login page (where a user can choose to login with external IDPs or directly with the broker). And after that point, the Dashboard is not involved.

Reply:
Yeah, this is a little loose detail that we need to nail down:  from the _perspective of a client application_, is there any difference between an identity _broker_ and an identity _provider_?  I'm really hoping that there isn't, as it seems like it would be a Bad Thing™.

It should be the case that we configure the Pbench Server with the URL to give to the client and that the client loads that URL into the browser when the user wants to authenticate...and it really shouldn't matter (to either the Dashboard or the Pbench Server) whether the user is talking directly to a provider or whether the user is talking to a broker (which redirects the user to a provider).  (In fact, it should be possible for a broker to redirect to what turns out to be _another broker_ which in turn redirects to a provider, and none of the parties up the chain should notice.)

> what Pbench-dashboard receives from the Pbench server is the URL for the Identity broker's authentication page, the identity broker may be configured with multiple external IDPs. When the user clicks the login button, the Dashboard causes the browser to load the Identity broker's login page

So, this is the particular deployment configuration which you are designing around...but, we should be able to use other configurations _without_ changing the Pbench Dashboard or Pbench Server **code**.

Reply:
@webbnh indeed, there is no difference between an identity broker and an identity provider from the client's perspective. I was just making sure it is clear to @portante that in this seq diagram they are shown as different entities for our team's understanding.

Reply:
Fair enough.  Indeed, to the extent that the sequence diagram depicts the full flow including both the broker and the provider, the distinction is important in that context.

However, I want to make sure that, in the context of the Pbench Server and Dashboard implementations, there is no distinction (and, even, no awareness that the thing might be a broker instead of a provider).